Drop "base URI" references

chrysn opened the pull request:
All URIs of RD APIs are discovered standalone, and no relative URI or path-appending is required any more. Therefore, the term "Base URIs" and the distinction between "rd-lookup-base" and "lookup-type" in the RD lookup API are obsolete.

This has already been [mentioned on the mailing list](https://mailarchive.ietf.org/arch/msg/core/7zzSmZK6Od1Rw3vzpaenN4G6otw), and [acked by Peter](https://mailarchive.ietf.org/arch/msg/core/NCLNlY8N__TCf9UMEq5yF1WxSJs), if I understood his response correctly (which I'm unsure about, thus tagging this for review).

chrysn commented:
I'd consider toning down the default value recommendations a different issue; is it fine with you apart from that?

(Note to self: I accidentally dropped the "res:= " parameter here; that's another topic and shouldn't go in that commit.)

cabo commented:
There is still some weirdness here: The "URIs" here are still parameters for URI templates, not the actual URIs that will be used for accesses.

chrysn commented:
Good point. Going from `/{+rd}{?ep...}` to `{+rd}{?ep...}` should address that, I'll update accordingly.

chrysn commented:
I've updated the pull request to address the comments, and fixed some oversights of mine.

While the first commit has been modified, the changes should largely be review friendly in that the three last commits are the newly introduced ones for
* @cabo's template weirdness
* @petervanderstok's "recommended" → "example" rewording and
* some more "base" occurrences I had missed, together with clearer naming of the "RD registration URI" instead of just "RD (base) URI" which might cause confusion with the "RD URIs" (the "RD lookup URIs" and the "RD group URI") in general.

chrysn commented:
Merged as agreed in last call.
